libvirt/resource_libvirt_domain.go

```diff
@@ -528,6 +535,76 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	// network-installation used in combination with kernel and initrd parameters.
+	NetworkAutoinstallEnabled := d.Get("network_autoinstall").(bool)
+	if NetworkAutoinstallEnabled {
```

Comment:
I think it would be better to have all the autoinstall related code in a separate file?

Reply:
If you have name suggestion feel free :)

Reply:
I would choose the name depending how we end merging this.

So, either _domain_reboot.go_ or _domain_autoinstall.go_....